languagetool-core/src/main/java/org/languagetool/rules/patterns/FalseFriendRuleHandler.java

```diff
@@ -96,6 +96,9 @@ public void startElement(String namespaceURI, String lName,
       if (Languages.isLanguageSupported(languageStr)) {
         Language tmpLang = Languages.getLanguageForShortCode(languageStr);
         currentTranslationLanguage = tmpLang;
+        if (currentTranslationLanguage == language) {
+          throw new RuntimeException("Translation language (" + currentTranslationLanguage + ") must not be the same as pattern language (" + language + ") for rule " + id);
+        }
```

Comment:
Wouldn't it be better to test this in a test instead of throwing a RuntimeException (which prevents the server from starting)?

Reply:
Yes, let me try it that way